modules/apps/message-boards/message-boards-service/src/main/java/com/liferay/message/boards/service/impl/MBCategoryLocalServiceImpl.java

```diff
@@ -105,6 +106,7 @@ public MBCategory addCategory(
 		MBCategory category = mbCategoryPersistence.create(categoryId);
 
 		category.setUuid(serviceContext.getUuid());
+		category.setExternalReferenceCode(externalReferenceCode);
```

Comment:
Maybe we have to validate here if exists a MBCategory with the externalReferenceCode as we are doing in other apis like https://github.com/liferay/liferay-portal/blob/f835ef37fd8e4f9ae58c7d54c484be1669ef6614/portal-impl/src/com/liferay/portlet/documentlibrary/service/impl/DLFileEntryLocalServiceImpl.java#L3987 or https://github.com/liferay/liferay-portal/blob/f1866a70a443e0d094d147bdcae5d0dbb716b689/modules/apps/journal/journal-service/src/main/java/com/liferay/journal/internal/validation/JournalArticleModelValidator.java#L516

Reply:
After adding the ERC to the model XML, the generated code is already checking for duplication. Anyway, I've added a test to check it in [a228cb3](https://github.com/liferay-content-management/liferay-portal/pull/5866/commits/a228cb3c261e0f2bfe8de7b598198b1f9aea47a9)

Reply:
oh, I didn't know it. Thanks for letting me know